src/transformers/modeling_utils.py

```diff
@@ -1179,7 +1173,9 @@ def _generate_beam_search(
         beam_scores = beam_scores.view(-1)  # shape (batch_size * num_beams,)
 
         # cache compute states
-        past = encoder_outputs  # defined for encoder-decoder models, None for decoder-only models
+        past = (
+            (encoder_outputs, None) if encoder_outputs is not None else None
+        )  # defined for encoder-decoder models, None for decoder-only models
```

Comment:
I ilke that idea here - that removes the unnecessary if else statements in the `prepare_inputs_for_generation` functions